### PR TITLE
[SeiDB] Pt 1: State Store Unit Tests

### PR DESCRIPTION
**Pull request by Kbhat1**
## Describe your changes and provide context
- Part 1 of State Store Unit Tests
- Sets of Storage Suite Test Framework + Initial tests (apply changeset, prune etc)
- NOTE: Also adds to pebble iterator immediate return + invalid iterator if requesting version earlier than prune height

## Testing performed to validate your change
- Verified test locally

### CODE REVIEW

File: ss/test/storage_test_suite.go [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+	}
+
+	itr, err := db.Iterator(storeKey1, 25, []byte("key000"), nil)
+	s.Require().NoError(err)
+	s.Require().False(itr.Valid())
+
+	// prune the latest version which should prune the entire dataset
+	s.Require().NoError(db.Prune(50))
+
+	for v := int64(1); v <= 50; v++ {
```
[the remark this answers is not in the document]
Comment (Kbhat1) [reply ss/test/storage_test_suite.go:542]: @yzang2019 It should leave the version (so not delete) but if you GET at that version of the prune height it should return nil regardless.

**Review by yzang2019 — APPROVED** [review]

File: ss/test/storage_test_suite.go [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+	s.Require().NoError(itr.Error())
+}
+
+func (s *StorageTestSuite) TestDatabaseIteratorNoDomain() {
+	db, err := s.NewDB(s.T().TempDir())
+	s.Require().NoError(err)
+	defer db.Close()
+
+	// for versions 1-50, set all 10 keys
+	for v := int64(1); v <= 50; v++ {
```
[the remark this answers is not in the document]
Comment (Kbhat1) [reply ss/test/storage_test_suite.go]: Updated

File: ss/test/storage_test_suite.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kbhat1) [reply ss/test/storage_test_suite.go:468]: Resolving this. In later PR part 2, going to add in another test where we prune and get at the height right after (which will return a value not nil)
